src/libcore/iter/mod.rs

```diff
@@ -590,7 +590,9 @@ unsafe impl<'a, I, T: 'a> TrustedRandomAccess for Cloned<I>
     }
 
     #[inline]
-    default fn may_have_side_effect() -> bool { true }
+    default fn may_have_side_effect() -> bool {
```

Comment:
This one is correct

Reply:
The one below should change to this, the specialization for Copy. Thanks for fixing! :heart:

Reply:
Thanks for noticing, I was going to fix it anyways, but I could have missed it, and then the situation would be even worse :+1:.